### DISCUSSION TITLE
Mostrar cuando nos pasamos de caracteres recomendados

### DISCUSSION THREAD

**Discussion by MontseTelefonica** [Telefonica/mistica-design#906] [Proposals]
Hola! Ha surgido la idea de, a raíz de no poder limitar caracteres pero sí recomendar una cantidad, la capacidad de mostrar en color rojo o en gris cuando nos excedemos de esa recomendación.

¿Esto es algo que tendría que definirse a nivel de componente desde Mistica? 

<img width="470" alt="image" src="https://user-images.githubusercontent.com/95479574/202143987-f42a9739-9578-4e37-bb0d-6a36b827c465.png">

**Comment by aweell** [comment]
Buenas @MontseTelefonica!


> a raíz de no poder limitar caracteres pero sí recomendar una cantidad

Actualmente sí se puede limitar el número de caracteres:

https://rebrand.ly/pvv7nq7


Esta discussion se había dado de forma interna pero nunca llegó a formalizarse, gracias por traerla aquí. My hot take here:

No impediría que el usuario escriba más caracteres de los permitidos, cuando se indica que el campo tiene una limitación de texto mostraría un contador en la parte inferior derecha del campo con la siguiente info:

(`n`) numero de caracteres introducido / (`y`) número de caracteres máximo

* Cuando `n` > `y` se mostrará `n` en rojo.
* Cuando el campo pierda el foco se mostrará en el mensaje de error que se ha superado el número de caracteres permitido

Ejemplo:

![Screenshot 2022-11-16 at 11 30 22](https://user-images.githubusercontent.com/44420072/202156784-387c60d7-5267-4114-89d4-f92c3f219a70.png)

**Comment by aweell** [comment]
Con respecto a superar el límite de caracteres en un inputField la idea inicial sería tener otra prop que se llame p.e `allowOverMax` que pueda estar a `true` o `false`. Siendo `false` por defecto para continuar con el comportamiento que tienen actualmente esos componentes.

Diferenciando dos casuísticas en el maxLenght:

* Hard max: no se permite superar el límite (Caso actual)
* Soft max: se permite superar el límite pero se muestra un error 

Al incluir un `maxLength`, tal y como ocurre ahora, el campo mostrará automáticamente el contador

Con respecto al contador, ya existe una issue de hecho que cubre esta problemática (#654). Archivo esta discussion
